src/Symfony/Bridge/Twig/Extension/TranslationExtension.php

```diff
@@ -91,12 +103,26 @@ public function getTranslationNodeVisitor(): TranslationNodeVisitor
         return $this->translationNodeVisitor ?: $this->translationNodeVisitor = new TranslationNodeVisitor();
     }
 
-    public function trans(string $message, array $arguments = [], string $domain = null, string $locale = null, int $count = null): string
+    /**
+     * @param string|Translatable $message The message id (may also be an object that can be cast to string)
+     */
+    public function trans($message, array $arguments = [], string $domain = null, string $locale = null, int $count = null): string
     {
+        if ($message instanceof Translatable) {
+            $arguments += $message->getParameters();
+            $domain = $message->getDomain();
```

Comment:
`$domain = $domain ?? $message->getDomain()`

Reply:
Not sure I agree with this one. Shouldn't the message know which domain it's part of better than the translator? If we did want to do something like this, maybe we want to allow setting the domain on `Translatable` to null, then update this to be something like `$domain = $message->getDomain() ?? $domain`.

This change would break situations like this:
```php
$message = t('error_message', [], 'errors);
```

```twig
{{ message|default('No Alerts')|trans(domain='messages') }}
```

Reply:
yep, i agree. I cant think of any case where you'd expect/want the domain to be overridden actually.

for parameters it does makes sense, as those are much more contextual